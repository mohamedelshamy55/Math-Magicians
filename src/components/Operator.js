import React from 'react';
import PropTypes from 'prop-types';

const Operator = ({ clickHandler, buttonName }) => (
  <>
    <li data-testid="oper-li"><button type="button" onClick={clickHandler}>{buttonName}</button></li>
  </>
);

export default Operator;

Operator.propTypes = {
  buttonName: PropTypes.string.isRequired,
  clickHandler: PropTypes.func.isRequired,
};
