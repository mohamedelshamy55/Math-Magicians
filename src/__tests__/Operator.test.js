import React from 'react';
import { render } from '@testing-library/react/';
import '@testing-library/jest-dom/extend-expect';
import Operator from '../components/Operator';

const renderer = require('react-test-renderer');

it('renders the props of component correctly', () => {
  const clickHandler = () => true;
  const buttonName = '+';
  const { getByTestId } = render(<Operator
    buttonName={buttonName}
    clickHandler={clickHandler}
  />);
  expect(getByTestId('oper-li')).toHaveTextContent('+');
});

it('matches snapshot 1 of Operator component', () => {
  const clickHandler = () => true;
  const buttonName = '-';
  const tree = renderer.create(<Operator
    buttonName={buttonName}
    clickHandler={clickHandler}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
