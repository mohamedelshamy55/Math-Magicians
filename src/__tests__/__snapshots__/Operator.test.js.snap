// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 of Operator component 1`] = `
<li
  data-testid="oper-li"
>
  <button
    onClick={[Function]}
    type="button"
  >
    -
  </button>
</li>
`;
